students/Chris_Hallen/lesson03/mailroom.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+
+response = ''
+donor_list = ['Zach Galifianakis', 23757, 3, "Conan O'Brien", 45000, 3, 'Kristen Wiig', 111222, 2, 'Sarah Silverman', 100, 1, 'Bill Burr', 12, 1, 'Will Ferrell', 100000, 2, ]
+
+while response != 'quit':  # This program will constantly run until a user enters 'quit' - regardless of whether 'quit'is upper or lower case or a combination thereof
```

Comment:
the code structure has couple issues:
* your entry point should be inside of `if __name__ == "__main__"` block (that would be your main while loop)
* as is now, you have one giant "block statement" - it's too complex (cyclomatic complexity)
* your functions should be defined on it's own (not inside a loop)
* your donor_list is defined at module level so no need to pass it into each function

Reply:
your rough code structure should look something like:
```
donors = [...]


def create_report():
    pass


def send_thank_you():
    pass


if __name__ == "__main__":
    while True:
        # call your functions
```